qiskit/circuit/qpy_serialization.py

```diff
@@ -247,9 +246,9 @@
         uint16_t num_parameters;
         uint32_t num_qargs;
         uint32_t num_cargs;
-        _Bool has_conditionl
-        uint16_t conditonal_reg_name_size;
-        long long conditional_value;
+        _Bool has_conditional;
+        uint16_t conditional_reg_name_size;
+        int64_t conditional_value;
```

Comment:
Unsure whether this value is signed or not. The ``conditional_value`` isn't documented in the docstring at all.

Reply:
Hmm, that's a good point this gets used with `c_if` which requires a non-negative integer so ideally this should have been unsigned. However, the struct uses a signed type: https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/circuit/qpy_serialization.py#L470-L483 (`q` is a signed 8 byte int: https://docs.python.org/3/library/struct.html#format-characters ) which means we can't really change this without potentially breaking people.

We could do it as part of the pending v2 format but I'm not sure it's worth the churn since it only matters for condition values that are larger than 2^63. Either way this is correct for now, if we do want to change it for QPY v2 we can do that in a separate PR.